File: app/src/main/java/com/karla/learningverbs/viewmodel/SplashViewModel.java
package com.karla.learningverbs.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.karla.learningverbs.repository.userrepository.UserRepository;
import com.karla.learningverbs.utils.base.BaseViewModel;

public class SplashViewModel extends BaseViewModel {
    private MutableLiveData<Boolean> isLoginStatus;
    private UserRepository splashScreenRepository;

    public SplashViewModel() {
        this.isLoginStatus = new MutableLiveData<>();
        this.splashScreenRepository = UserRepository.getInstance();
    }

    public void isLogin() {
        isLoginStatus.setValue(splashScreenRepository.isLogin());
    }

    public LiveData<Boolean> getIsLogin() {
        return isLoginStatus;
    }
}

File: app/src/main/java/com/karla/learningverbs/kotlin/viewmodel/SplashViewModel.kt
package com.karla.learningverbs.kotlin.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.karla.learningverbs.kotlin.repository.userrepository.UserRepository
import com.karla.learningverbs.kotlin.utils.base.BaseViewModel

class SplashViewModel : BaseViewModel() {

    private var isLoginStatus = MutableLiveData<Boolean>()
    private var splashScreenRepository: UserRepository = UserRepository()

    fun isLogin() {
        isLoginStatus.value = splashScreenRepository.isLogin()
    }

    fun getIsLogin(): LiveData<Boolean> {
        return isLoginStatus
    }
}